pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/item/builder/ItemStackBuilder.kt

```diff
@@ -150,6 +189,126 @@ open class ItemStackBuilder private constructor(val stack: ItemStack) : ItemProv
+
+    fun boots(
+        armor: Double,
+        armorToughness: Double
+    ) = armor(EquipmentSlotGroup.FEET, armor, armorToughness)
+
+    fun armor(
+        slot: EquipmentSlotGroup,
+        armor: Double,
+        armorToughness: Double
+    ) = apply {
+        removeAttributeModifiers(Attribute.ARMOR)
+        removeAttributeModifiers(Attribute.ARMOR_TOUGHNESS)
+        addAttributeModifier(Attribute.ARMOR, AttributeModifier(baseArmor, armor, AttributeModifier.Operation.ADD_NUMBER, slot))
+        addAttributeModifier(Attribute.ARMOR_TOUGHNESS, AttributeModifier(baseArmorToughness, armorToughness, AttributeModifier.Operation.ADD_NUMBER, slot))
+    }
+
+    fun axe(
+        miningSpeed: Float,
+        miningDurabilityDamage: Int
+    ) = tool(Registry.BLOCK.getTag(BlockTypeTagKeys.MINEABLE_AXE), miningSpeed, miningDurabilityDamage)
+
+    fun pickaxe(
+        miningSpeed: Float,
+        miningDurabilityDamage: Int
+    ) = tool(Registry.BLOCK.getTag(BlockTypeTagKeys.MINEABLE_PICKAXE), miningSpeed, miningDurabilityDamage)
+
+    fun shovel(
+        miningSpeed: Float,
+        miningDurabilityDamage: Int
+    ) = tool(Registry.BLOCK.getTag(BlockTypeTagKeys.MINEABLE_SHOVEL), miningSpeed, miningDurabilityDamage)
+
+    fun hoe(
+        miningSpeed: Float,
+        miningDurabilityDamage: Int
+    ) = tool(Registry.BLOCK.getTag(BlockTypeTagKeys.MINEABLE_HOE), miningSpeed, miningDurabilityDamage)
+
+    fun tool(
+        blocks: RegistryKeySet<BlockType>,
+        miningSpeed: Float,
+        miningDurabilityDamage: Int
+    ) = apply {
+        set(DataComponentTypes.TOOL, Tool.tool()
+            .defaultMiningSpeed(miningSpeed)
+            .damagePerBlock(miningDurabilityDamage)
+            .addRule(Tool.rule(blocks, miningSpeed, TriState.TRUE)))
+    }
+
+    fun noTool() = unset(DataComponentTypes.TOOL)
+
+    fun weapon(
+        attackDamage: Double,
+        attackSpeed: Double,
+        attackDurabilityDamage: Int,
+        disableShieldSeconds: Float
+    ) = apply {
+        addAttributeModifier(Attribute.ATTACK_DAMAGE, AttributeModifier(baseAttackDamage, -1.0 + attackDamage, AttributeModifier.Operation.ADD_NUMBER, EquipmentSlotGroup.MAINHAND))
+        addAttributeModifier(Attribute.ATTACK_SPEED, AttributeModifier(baseAttackSpeed, -4.0 + attackSpeed, AttributeModifier.Operation.ADD_NUMBER, EquipmentSlotGroup.MAINHAND))
+        set(DataComponentTypes.WEAPON, Weapon.weapon()
+            .itemDamagePerAttack(attackDurabilityDamage)
+            .disableBlockingForSeconds(disableShieldSeconds))
+    }
+
+    fun attackKnockback(knockback: Double) = apply {
+        removeAttributeModifiers(Attribute.ATTACK_KNOCKBACK)
+        addAttributeModifier(Attribute.ATTACK_KNOCKBACK, AttributeModifier(baseAttackKnockback, knockback, AttributeModifier.Operation.ADD_NUMBER, EquipmentSlotGroup.MAINHAND))
+    }
+
+    fun durability(durability: Int) = set(DataComponentTypes.MAX_DAMAGE, durability)
```

Comment:
```suggestion
    
    fun durability(durability: Int) = durability(durability, 0)

    fun durability(durability: Int, damage: Int) = apply {
        set(DataComponentTypes.MAX_DAMAGE, durability)
        set(DataComponentTypes.DAMAGE, damage)
    }
```
if the item doesn't have a durability by default, durability bar won't work since the item has no damage. eg reactivated wither skull